frame/vesting/src/lib.rs

```diff
@@ -302,72 +409,273 @@ pub mod pallet {
+		)]
 		pub fn force_vested_transfer(
 			origin: OriginFor<T>,
 			source: <T::Lookup as StaticLookup>::Source,
 			target: <T::Lookup as StaticLookup>::Source,
 			schedule: VestingInfo<BalanceOf<T>, T::BlockNumber>,
 		) -> DispatchResult {
 			ensure_root(origin)?;
-			ensure!(schedule.locked >= T::MinVestedTransfer::get(), Error::<T>::AmountLow);
-
-			let target = T::Lookup::lookup(target)?;
-			let source = T::Lookup::lookup(source)?;
-			ensure!(!Vesting::<T>::contains_key(&target), Error::<T>::ExistingVestingSchedule);
+			Self::do_vested_transfer(source, target, schedule)
+		}
 
-			T::Currency::transfer(&source, &target, schedule.locked, ExistenceRequirement::AllowDeath)?;
+		/// Merge two vesting schedules together, creating a new vesting schedule that unlocks over
+		/// highest possible start and end blocks. If both schedules have already started the current 
+		/// block will be used as the schedule start; with the caveat that if one schedule is finishes by 
+		/// the current block, the other will be treated as the new merged schedule, unmodified.
+		///
+		/// NOTE: If `schedule1_index == schedule2_index` this is a no-op.
+		/// NOTE: This will unlock all schedules through the current block prior to merging.
+		/// NOTE: If both schedules have ended by the current block, no new schedule will be created.
+		///
+		/// Merged schedule attributes:
+		/// starting_block: `MAX(schedule1.starting_block, scheduled2.starting_block, current_block)`.
+		/// ending_block: `MAX(schedule1.ending_block, schedule2.ending_block)`.
+		/// locked: `schedule1.locked_at(current_block) + schedule2.locked_at(current_block)`.
+		///
+		/// The dispatch origin for this call must be _Signed_.
+		///
+		/// - `schedule1_index`: index of the first schedule to merge.
+		/// - `schedule2_index`: index of the second schedule to merge.
+		///
+		/// # <weight>
+		/// - `O(1)`.
+		/// - DbWeight: TODO Reads, TODO Writes
+		///     - Reads: TODO
+		///     - Writes: TODO
+		/// # </weight>
+		#[pallet::weight(
+			T::WeightInfo::not_unlocking_merge_schedules(MaxLocksOf::<T>::get())
+			.max(T::WeightInfo::unlocking_merge_schedules(MaxLocksOf::<T>::get()))
+		)]
+		pub fn merge_schedules(
+			origin: OriginFor<T>,
+			schedule1_index: u32,
+			schedule2_index: u32,
+		) -> DispatchResult {
+			if schedule1_index == schedule2_index {
+				return Ok(());
+			};
+			let who = ensure_signed(origin)?;
+			let schedule1_index = schedule1_index as usize;
+			let schedule2_index = schedule2_index as usize;
+			let vesting = Self::vesting(&who).ok_or(Error::<T>::NotVesting)?;
+			let len = vesting.len();
+			ensure!(
+				schedule1_index < len && schedule2_index < len,
+				Error::<T>::ScheduleIndexOutOfBounds
+			);
+
+			// The schedule index is based off of the schedule ordering prior to filtering out any
+			// schedules that may be ending at this block.
+			let schedule1 = vesting[schedule1_index];
```

Comment:
You could get rid of the explicit ensure above by writing this as 

```suggestion
			let schedule1 = vesting.get(&schedule1_index).ok_or(Error::<T>::ScheduleIndexOutOfBounds)?;
```